ibm/service/project/data_source_ibm_project.go

```diff
@@ -0,0 +1,812 @@
+// Copyright IBM Corp. 2023 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package project
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/project-go-sdk/projectv1"
+)
+
+func DataSourceIbmProject() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIbmProjectRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
```

Comment:
I don't like to make manual changes to the generated code, our friend @dpopp07 can explain better how the OpenAPI SDK generator works, maybe some directives/annotations exist to do this automatically

Reply:
this is our API spec https://github.ibm.com/ibmcloud/projects/blob/main/services/projects-api/api/swagger.yaml, the terraform code is generate automatically from API spec

Reply:
Hmm I would've thought the generator would not produce an `id` field in the schema but that is a part of our code that currently needs some work. I'll look into it

Reply:
we have a private slack chat to discuss this

Reply:
Okay, we're close to having the generator fix in place for this.

@hkantare I do have one question: Would it be more idiomatic to drop ID field altogether in favor of using the implicit `id` field, since this resource has no parent resources? Or is it more consistent with other providers to use `project_id` ?

Reply:
code regenerated